ant/src/org/jetbrains/kotlin/ant/Kotlin2JvmTask.kt

```diff
@@ -73,4 +81,43 @@ class Kotlin2JvmTask : KotlinCompilerBaseTask() {
         if (noReflect) args.add("-no-reflect")
         if (includeRuntime) args.add("-include-runtime")
     }
+
+    override fun execute() {
+        if (!fork)
+            super.execute()
+        else {
+            val compilerClass = KotlinAntTaskUtil.getOrCreateClassLoader().loadClass(compilerFqName)
+            val jvmArgs = cmdl.vmCommand.arguments
+            exec(compilerClass, jvmArgs)
+        }
+    }
+
+    private fun exec(clazz: Class<*>, jvmArgs: Array<String>): Int {
+        val javaHome = System.getProperty("java.home")
+        val javaBin = javaHome + File.separator + "bin" + File.separator + "java"
+        val classpath = System.getProperty("java.class.path")
+        val className = clazz.name
+
+        val command = ArrayList<String>()
+        command.add(javaBin)
+        command.addAll(jvmArgs)
+        command.add("-cp")
+        command.add(classpath + File.separator + compileClasspath.toString())
```

Comment:
Did you mean `pathSeparator` here?

Reply:
Yes, the `compileClasspath.toString()` is `<pathelement location="${lib}/kotlinc/lib/kotlin-compiler.jar"/>` in my case.